### PR TITLE
poppler: 21.05.0 -> 21.06.01

### LINKED ISSUE #126685: Poppler out of date, evince to crash
###### Checklist

- [x] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name: nixos.poppler

current version: 21.05.0
desired version: 21.06.01

###### Notify maintainers

maintainers: @ttuegel @jtojnar 

###### Note for maintainers

Please tag this issue in your PR.

The current version of `poppler` is causing evince to crash frequently:
https://gitlab.gnome.org/GNOME/evince/-/issues/1608#note_1136600

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Motivation for this change
https://lists.freedesktop.org/archives/poppler/2021-June/014771.html
https://lists.freedesktop.org/archives/poppler/2021-June/014772.html

Closes #126685 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 127832` at 004970b1 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages marked as broken and skipped:</summary>
  <ul>
    <li>grass</li>
    <li>green-pdfviewer</li>
    <li>hplipWithPlugin_3_16_11</li>
    <li>hplipWithPlugin_3_18_5</li>
    <li>hplip_3_16_11</li>
    <li>hplip_3_18_5</li>
    <li>odpdown</li>
    <li>python38Packages.pygmt</li>
    <li>python39Packages.pygmt</li>
    <li>qlandkartegt</li>
    <li>spring</li>
    <li>springLobby</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/3c7d75de28fba09868e85a825ac6ba35">libsForQt512.kfilemetadata</a></li>
  </ul>
</details>
<details>
  <summary>314 packages skipped due to time constraints:</summary>
  <ul>
    <li>adapta-gtk-theme</li>
    <li>apacheHttpdPackages.mod_tile (apacheHttpdPackages_2_4.mod_tile)</li>
    <li>apvlv</li>
    <li>arc-theme</li>
    <li>arx-libertatis</li>
    <li>asciidoc-full</li>
    <li>asciidoc-full-with-plugins</li>
    <li>bibata-cursors</li>
    <li>bibata-cursors-translucent</li>
    <li>bibata-extra-cursors</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>55 packages built successfully:</summary>
  <ul>
    <li>aesop</li>
    <li>anystyle-cli</li>
    <li>colord</li>
    <li>colord-gtk</li>
    <li>diff-pdf</li>
    <li>epkowa</li>
    <li>fbida</li>
    <li>fgallery</li>
    <li>gnome.gnome-color-manager</li>
    <li>gnome.gnome-settings-daemon (gnome.gnome_settings_daemon)</li>
    <li>gnome.gnome-user-share</li>
    <li>gnome.nautilus</li>
    <li>tracker-miners (gnome.tracker-miners)</li>
    <li>haskellPackages.bindings-sane</li>
    <li>haskellPackages.gi-poppler</li>
    <li>leela</li>
    <li>libsForQt5.baloo (libsForQt515.baloo ,plasma5Packages.baloo)</li>
    <li>libsForQt5.baloo-widgets (libsForQt515.baloo-widgets ,plasma5Packages.baloo-widgets)</li>
    <li>libsForQt5.kfilemetadata (libsForQt515.kfilemetadata ,plasma5Packages.kfilemetadata)</li>
    <li>libsForQt5.poppler (libsForQt515.poppler ,plasma5Packages.poppler)</li>
    <li>libsForQt5.poppler_0_61 (libsForQt515.poppler_0_61 ,plasma5Packages.poppler_0_61)</li>
    <li>libsForQt512.poppler</li>
    <li>libsForQt512.poppler_0_61</li>
    <li>libsForQt514.baloo</li>
    <li>libsForQt514.baloo-widgets</li>
    <li>libsForQt514.kfilemetadata</li>
    <li>libsForQt514.poppler</li>
    <li>libsForQt514.poppler_0_61</li>
    <li>pdf-quench</li>
    <li>pdf2svg</li>
    <li>pdfgrep</li>
    <li>pdfmod</li>
    <li>pdfsandwich</li>
    <li>pdftag</li>
    <li>perl530Packages.ImageSane</li>
    <li>perl532Packages.ImageSane</li>
    <li>poppler</li>
    <li>poppler_0_61</li>
    <li>poppler_gi</li>
    <li>poppler_min</li>
    <li>poppler_utils</li>
    <li>pqiv</li>
    <li>pympress</li>
    <li>python38Packages.openpaperwork-gtk</li>
    <li>python38Packages.pdf2image</li>
    <li>python38Packages.pdftotext</li>
    <li>python38Packages.sane</li>
    <li>python39Packages.openpaperwork-gtk</li>
    <li>python39Packages.pdf2image</li>
    <li>python39Packages.pdftotext</li>
    <li>python39Packages.sane</li>
    <li>sane-backends</li>
    <li>sane-frontends</li>
    <li>scanbd</li>
    <li>setzer</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/development/libraries/poppler/0.61.nix [hunk 1]
```diff
@@ -12,25 +27,38 @@ let
   mkFlag = optset: flag: "-DENABLE_${flag}=${if optset then "on" else "off"}";
 in
 stdenv.mkDerivation rec {
-  name = "poppler-${suffix}-${version}";
+  pname = "poppler-${suffix}";
+  inherit version;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/poppler/0.61.nix:31]: Why not just move version to here?

File: pkgs/development/libraries/poppler/0.61.nix [hunk 2]
```diff
@@ -12,25 +27,38 @@ let
...
+    cmake
+    ninja
+    pkg-config
+  ];
+
+  buildInputs = [
+    libiconv
+    libintl
+  ]
+  ++ lib.optional withData poppler_data;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/poppler/0.61.nix:61]:
```suggestion
  ] ++ lib.optional withData poppler_data;
```

File: pkgs/development/libraries/poppler/default.nix [hunk 3]
```diff
@@ -1,44 +1,46 @@
...
+, ninja
 , openjpeg
+, pkg-config
+, scribusUnstable
+, texlive
+, zlib
 , withData ? true, poppler_data
 , qt5Support ? false, qtbase ? null
 , introspectionSupport ? false, gobject-introspection ? null
 , utils ? false, nss ? null
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/poppler/default.nix:27]:
```suggestion
, qt5Support ? false, qtbase
, introspectionSupport ? false, gobject-introspection
, utils ? false, nss
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by hmenke** [comment]
Any chance this could be backported? Evince on NixOS 21.05 still regularly crashes.

**Comment by hmenke** [comment]
Could we maybe just backport the entire PR? The constant segfaults are becoming really annoying.

**Pull request merged**